src/Sylius/Bundle/ApiBundle/Serializer/AdjustmentNormalizer.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Sylius\Bundle\ApiBundle\Serializer;
+
+use Sylius\Bundle\ApiBundle\Helpers\CustomMoneyTransformerInitializer;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+
+class AdjustmentNormalizer implements ContextAwareNormalizerInterface, NormalizerAwareInterface
+{
+    use NormalizerAwareTrait;
+
+    private const ALREADY_CALLED = 'adjustment_normalizer_already_called';
+
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        $context[self::ALREADY_CALLED] = true;
+
+        $data = $this->normalizer->normalize($object, $format, $context);
+        if (isset($data['amount'])) {
+            $transformer = CustomMoneyTransformerInitializer::getPriceTransformer();
```

Comment:
We shouldn't create a new instance of `SyliusMoneyTransformer` via `CustomMoneyTransformerInitializer`. We must inject `SyliusMoneyTransformer` service and use it directly without a new layer